src/app/fdctl/run/tiles/fd_resolv.c

```diff
@@ -167,10 +254,18 @@ unprivileged_init( fd_topo_t *      topo,
   ctx->round_robin_cnt = fd_topo_tile_name_cnt( topo, tile->name );
   ctx->round_robin_idx = tile->kind_id;
 
+  ctx->completed_slot = 0UL;
+  ctx->blockhash_ring_idx = 0UL;
+
   if( FD_LIKELY( !tile->kind_id ) ) _fd_ext_resolv_tile_cnt = ctx->round_robin_cnt;
 
   ctx->root_bank = NULL;
 
+  memset( &ctx->metrics, 0, sizeof( ctx->metrics ) );
```

Comment:
I think you should zero out the ring as well. I know for prod it will be in pre-zeroed pages, but it's safer for the first 4k slots when you're deleting a blockhash based on uninitialized memory.